src/vantage-dimmer-accessory.ts

```diff
@@ -0,0 +1,180 @@
+  name: string;
+
+  private readonly lightService: Service;
+  private readonly informationService: Service;
+
+  // TODO: dont need loadType, but because of rgb it is kept.
+  constructor(hap: HAP, log: Logging, name: string, vid: string, controller: VantageInfusionController, loadType: string) {
+    this.log = log;
+    this.hap = hap;
+    this.name = name;
+    this.vid = vid;
+    this.controller = controller;
+    this.brightness = 100;
+    this.saturation = 0;
+    this.hue = 0;
+    this.loadType = loadType;
+    this.dimmerRequestTimer = undefined;
+
+    this.lightService = new hap.Service.Lightbulb(name);
+    this.addLightService();
+
+    this.informationService = new hap.Service.AccessoryInformation()
+      .setCharacteristic(hap.Characteristic.Manufacturer, "Vantage Controls")
+      .setCharacteristic(hap.Characteristic.Model, "Power Switch Dimmer")
+      .setCharacteristic(hap.Characteristic.SerialNumber, `VID ${this.vid}`);
+
+    // get the current state
+    this.controller.sendGetLoadStatus(this.vid);
+  }
+
+  addLightService() {
+    this.lightService.getCharacteristic(this.hap.Characteristic.On)
+      .on(CharacteristicEventTypes.GET, (callback: CharacteristicGetCallback) => {
+        this.log.debug(`Dimmer ${this.name} get state: ${this.lightOn ? "ON" : "OFF"}`);
+        callback(undefined, this.lightOn);
+      })
+      .on(CharacteristicEventTypes.SET, (value: CharacteristicValue, callback: CharacteristicSetCallback) => {
+        this.log.debug(`Dimmer ${this.name} set state: ${value ? "ON" : "OFF"}`);
+        this.lightOn = value as boolean;
+        callback();
+      });
+
+    if (this.loadType == "dimmer" || this.loadType == "rgb") {
+      this.addDimmerLightService();
+    }
+
+    if (this.loadType == "rgb") {
+      this.addRGBLightService();
+    }
+  }
+
+  dispatchDimmerRequest() {
+    this.dimmerRequestTimer = setTimeout(() => {
+      this.controller.sendLoadDim(this.vid, this.lightOn ? this.brightness : 0);
+      this.dimmerRequestTimer = undefined;
+    }, 50);
+  }
+
+  /*
+   * adds dimmer control to light service
+   */
+  addDimmerLightService() {
+    this.lightService.getCharacteristic(this.hap.Characteristic.Brightness)
+      .on(CharacteristicEventTypes.GET, (callback: CharacteristicGetCallback) => {
+        this.log.debug(`Dimmer ${this.name} get brightness state: ${this.brightness}`);
+        callback(undefined, this.brightness);
+      })
+      .on(CharacteristicEventTypes.SET, (value: CharacteristicValue, callback: CharacteristicSetCallback) => {
+        this.log.debug(`Dimmer ${this.name} set brightness state: ${value}`);
+        this.brightness = value as number;
+        this.lightOn = this.brightness > 0 ? true : false;
+
+        if (this.dimmerRequestTimer == undefined) {
+          this.dispatchDimmerRequest();
+
+        } else {
+          clearTimeout(this.dimmerRequestTimer);
+          this.dispatchDimmerRequest();
+        }
```

Comment:
As far as I can tell this is a debouncing mechanism (like https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940)

just document it

Reply:
oh cool was looking for how to do this, didn't know it was called debouncing.